src/v2/cookbook/electron-vue-cookbook.md

```diff
@@ -0,0 +1,705 @@
+<script>
+export default {
+  name: "App",
+  data() {
+    return {
+      minutes: 1,
+      seconds: 0,
+      // add the timer variable
+      timer: null
+    };
+  },
+  // add the methods
+  methods: {
+    startTimer() {
+      this.timer = setInterval(() => this.countdown(), 1000);
+    },
+    resetTimer() {
+      clearInterval(this.timer);
+      this.timer = null;
+      this.minutes = 1;
+      this.seconds = 0;
+    },
+    countdown() {
+      this.seconds--;
+      if (this.seconds < 0) {
+        this.minutes--;
+        if (this.minutes < 0) {
+          this.resetTimer();
+        } else {
+          this.seconds = 59;
+        }
+      }
+    }
+  },
+  computed: {
+    minutesText() {
+      return String(this.minutes).padStart(2, "0");
+    },
+    secondsText() {
+      return String(this.seconds).padStart(2, "0");
+    }
+  }
+};
+</script>
+```
+
+Now you should be able to start and reset the timer. You might notice the timer counts down faster when you click Start several times while the timer is already running. To avoid this from happening, let's disable the Start button when the timer is already running. We'll add a variable called `running` which will represent the state of the timer and use that variable to disable the Start button while the timer is running. Make the necessary changes indicated in the comments below.
+
+```vue
+<template>
+<div>
+  
+  <h1>{{minutesText}}:{{secondsText}}</h1>
+
+  <!--Use the running variable to disable buttons-->
+  <button @click="startTimer" :disabled="running">Start</button>
+  <button @click="resetTimer" :disabled="!running">Reset</button>
+
+</div>    
+</template>
+
+<script>
+export default {
+  name: "App",
+  data() {
+    return {
+      minutes: 1,
+      seconds: 0,
+      timer: null,
+      running: false
+    };
+  },
+  methods: {
+    startTimer() {
+      this.timer = setInterval(() => this.countdown(), 1000);
+
+      // flip the running variable to true
+      this.running = true;
+    },
+    resetTimer() {
+      clearInterval(this.timer);
+      this.timer = null;
+      this.minutes = 1;
+      this.seconds = 0;
+
+      // flip the running variable to false
+      this.running = false;
+    },
+    countdown() {
+      this.seconds--;
+      if (this.seconds < 0) {
+        this.minutes--;
+        if (this.minutes < 0) {
+          this.resetTimer();
+        } else {
+          this.seconds = 59;
+        }
+      }
+    }
+  },
+  computed: {
+    minutesText() {
+      return String(this.minutes).padStart(2, "0");
+    },
+    secondsText() {
+      return String(this.seconds).padStart(2, "0");
+    }
+  }
+};
+</script>
+
+<style>
+</style>
+```
+
+Next, let's add buttons for setting the timer's default value. We'll need to add variables for the default minutes and seconds too. When we click the Reset button, our timer will set itself to the default minutes and seconds variables. Make the changes indicated in the comments below.
+
+```vue<template>
```

Comment:
This needs to be on a separate line, like:
```
... vue
<template>
```
(can't put backticks in because it would break, but hopefully you get the drift)